protocols/client/client.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// gRPC client wrapper
+
+package client
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/mdlayher/vsock"
+	"google.golang.org/grpc"
+
+	agentgrpc "github.com/kata-project/agent/protocols/grpc"
+)
+
+const (
+	unixSocketScheme  = "unix"
+	vsockSocketScheme = "vsock"
+	dialTimeout       = 5 * time.Second
+)
+
+type AgentClient struct {
```

Comment:
Since this is going to be an exported package, could we document all exported structures and methods:

* `AgentClient`
* `NewAgentClient()`
* `AgentClient.Close()`